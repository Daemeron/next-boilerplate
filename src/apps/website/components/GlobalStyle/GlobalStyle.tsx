import { createGlobalStyle } from 'styled-components';
import { normalize } from 'polished';

/**
 * TODO: Write tests for <GlobalStyle /> component
 * BODY: Currently react-test-renderer cannot render components created with createGlobalStyle.
 *       When this behaviour get's patched, unit test should be written.
 */
export const GlobalStyle = createGlobalStyle`
  * {
    padding: 0;
    margin: 0;
    outline: 0;
    box-sizing: border-box;
    font-size: inherit;
    font-weight: inherit;
  }
  body {
    font-weight: normal;
    color: #000;
    background-color: #fff;
  }
  ${normalize()}
`;
