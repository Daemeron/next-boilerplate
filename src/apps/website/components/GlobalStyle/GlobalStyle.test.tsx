import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { GlobalStyle } from './GlobalStyle';

describe('GlobalStyle', () => {
  test('matches snapshot', () => {
    const globalStyle = render(<GlobalStyle />);
    expect(globalStyle.container).toMatchSnapshot();
  });
});
