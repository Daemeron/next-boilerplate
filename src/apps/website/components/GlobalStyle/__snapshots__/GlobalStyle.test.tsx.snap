// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStyle matches snapshot 1`] = `<div />`;
